apps/dashboard/app/(app)/settings/billing/components/usage-insights.tsx

```diff
@@ -0,0 +1,233 @@
+            <div className="z-10">
+              <div className="flex items-center justify-between">
+                <div className="flex items-center justify-start gap-2">
+                  <h2 className="text-gray-12 text-lg capitalize">
+                    {dataToUse.plan}
+                  </h2>
+                </div>
+
+                {dataToUse.current / dataToUse.max >= 0.94 ? (
+                  <Button variant="primary" size="sm">
+                    <Link href="/settings/billing">Upgrade</Link>
+                  </Button>
+                ) : (
+                  <Button variant="outline" size="sm">
+                    <Link href="/settings/billing">Manage Plan</Link>
+                  </Button>
+                )}
+              </div>
+
+              {children}
+            </div>
+          </motion.div>
+        )}
+      </UsageProvider>
+    )
+  }
+)
+
+Root.displayName = ROOT_NAME
+
+/* ----------------------------------------------------------------------------
+ * UsageInsight - Details
+ * --------------------------------------------------------------------------*/
+
+const DETAILS_NAME = "UsageDetails"
+
+export const Details = React.forwardRef<DivElement, DivProps>((props, ref) => {
+  const { className, children, ...detailsProps } = props
+
+  return (
+    <motion.div
+      initial={{ opacity: 0, x: -16 }}
+      animate={{ opacity: 1, x: -0 }}
+      transition={{
+        ease: "easeInOut",
+        duration: 0.4,
+        opacity: { duration: 0.8, delay: 0.1 },
+      }}
+      {...detailsProps}
+      ref={ref}
+      className={cn(
+        "h-16 duration-500 ease-out w-full flex flex-col items-start justify-start pt-4",
+        {
+          className,
+        }
+      )}
+    >
+      {children}
+    </motion.div>
+  )
+})
+
+Details.displayName = DETAILS_NAME
+
+/* ----------------------------------------------------------------------------
+ * UsageInsight - Item
+ * --------------------------------------------------------------------------*/
+type PrimitiveDivElement = React.ElementRef<"div">
+type PrimitiveDivProps = React.ComponentPropsWithoutRef<"div">
+
+export interface UsageItemProps extends PrimitiveDivProps {
+  item?: {
+    current: number
+    max: number
+  }
+  title: string
+  description?: string
+  color?: string
+}
+
+const ITEM_NAME = "UsageItem"
+
+export const Item = React.forwardRef<PrimitiveDivElement, UsageItemProps>(
+  (props, ref) => {
+    const { item, title, description, color, className, ...itemProps } = props
+    const { format } = Intl.NumberFormat(undefined, { notation: "compact" })
+    const { current, max } = useUsageContext("UsageItem")
+
+    return (
+      <div
+        {...itemProps}
+        ref={ref}
+        className={cn("flex items-start justify-start gap-3", {
+          className,
+        })}
+      >
+        <ProgressCircle
+          value={item?.current ?? current}
+          max={item?.max ?? max}
+          color={color ?? "#f76e19"}
```

Comment:
Please, let's use colors from our palette. For this one you can probably use `orange`.